R/source.R

```diff
@@ -46,13 +52,50 @@ source_file <- function(path, env = test_env(), chdir = TRUE,
       error = function(err) {
         abort(
           paste0("In path: ", encodeString(path, quote = '"')),
-          parent = err
+          parent = err,
+          call = error_call
         )
       }
     )
   }
 }
 
+filter_label <- function(exprs, label = NULL, error_call = caller_env()) {
+  if (is.null(label)) {
+    return(exprs)
+  }
+
+  found <- FALSE
+  include <- rep(FALSE, length(exprs))
+
+  for (i in seq_along(exprs)) {
+    expr <- exprs[[i]]
+
+    if (!is_call(expr, "test_that", n = 2)) {
```

Comment:
```suggestion
    if (!is_call(expr, "test_that", n = 2) && !found) {
```

What if we only execute top-level code _before_ the matching `test_that()` call?

Reply:
Oooh good thought.